example/test_driver/cloud_firestore_behaviors.dart

```diff
@@ -0,0 +1,92 @@
+import 'dart:async';
+
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:cloud_firestore_mocks/cloud_firestore_mocks.dart';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter_driver/driver_extension.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:test/test.dart' as _test;
+
+Future<Firestore> createFireStoreClient(
+    String appName, String host, bool sslEnabled) async {
+  // This is from https://github.com/FirebaseExtended/flutterfire/blob/master/packages/cloud_firestore/cloud_firestore/example/test_driver/cloud_firestore.dart
+  final FirebaseOptions firebaseOptions = const FirebaseOptions(
+    googleAppID: '1:79601577497:ios:5f2bcc6ba8cecddd',
+    gcmSenderID: '79601577497',
+    apiKey: 'AIzaSyArgmRGfB5kiQT6CunAOmKRVKEsxKmy6YI-G72PVU',
+    projectID: 'flutter-firestore',
+  );
+  final FirebaseApp app = await FirebaseApp.configure(
+    name: appName,
+    options: firebaseOptions,
+  );
+  final firestore = Firestore(app: app);
+  await firestore.settings(
+    persistenceEnabled: true,
+    host: host,
+    sslEnabled: sslEnabled,
+    cacheSizeBytes: 1048576,
+  );
+  return firestore;
+}
+
+/// Test case to compare 3 Firestore implementation behaviors: Cloud
+/// Firestore backend, Emulator backend, and cloud_firestore_mocks.
+void main() {
+  final Completer<String> completer = Completer<String>();
+  enableFlutterDriverExtension(handler: (_) => completer.future);
+  tearDownAll(() => completer.complete(null));
+
+  group('Firestore behavior comparison:', () {
+    Map<String, Future<Firestore>> firestoreFutures = {
```

Comment:
I had to make the value futures, because we need to call 'test' function without async.

If the value were without futures, it would have been initialized somewhere in setup method using future/async. When I tried that Flutter did not recognize test function calls.